### PR TITLE
Add unit tests and refactor some of the model code

### PR DESCRIPTION
**Pull request by hwayne**
This pull request makes the following changes:

* Unit tests for `lesson` and `code`. Create testing database with `rake db:test:prepare`, run with `rake test`.
* Some refactoring of the sorting and scoping logic, mostly to reduce duplication. For example, all new sorting methods will now automatically support `desc` without having to modify their `where` clauses.
* A couple of minor bugfixes
* Now works with any version of Ruby above 2.1.

### CODE REVIEW

**Comment by hwayne** [comment]
Since the searching logic is pretty gnarly I thought it'd be a good idea to show that it still works properly
screen shot 2016-02-23 at 11 23 00 pm

File: edumap/app/models/lesson.rb [hunk 1]
```diff
@@ -10,61 +10,48 @@ class Lesson < ActiveRecord::Base
...
-      terms.map {
-        or_clauses = [
-          "LOWER(lessons.name) LIKE ?",
-          "LOWER(lessons.lesson_url) LIKE ?",
-        ].join(' OR ')
-        "(#{ or_clauses })"
-      }.join(' AND '),
-      *terms.map { |e| [e] * num_or_conditions }.flatten
-    )
+                 .map { |e| "%#{e.gsub('*', '%')}%" }
```
Comment (hwayne) [inline edumap/app/models/lesson.rb]: Honestly this should probably be split into `.map {|e| e.gsub /*/, /%/}.map{|e| "%#{e}%"}` to make it a bit clearer that it's two different processings

File: edumap/app/models/code.rb [hunk 2]
```diff
@@ -12,35 +12,30 @@ class Code < ActiveRecord::Base
   belongs_to :standard
   has_and_belongs_to_many :lessons
 
-  scope :sorted_by, lambda { |sort_option|
+  scope :with_standard_id, -> (ids) { where(standard_id: ids) }
+  scope :with_created_at_gte, -> (ref_date) { where('created_at >= ?', ref_date) }
```
Comment (hwayne) [inline edumap/app/models/code.rb]: Note this is shared with `lesson`; I wonder if it'd make sense to make add a filtering concern that has this and the basic skeleton of `sorted_by` (specifically the logic for errors and reversing the order)

**Comment by pjsier** [comment]
Thanks for taking this on! I just tried it out and it works great, feel free to post any issues of other things you think we need.

**Pull request merged**